oauth_manager.go

```diff
@@ -554,16 +557,42 @@ func (r *RedisOsinStorageInterface) GetClientTokens(id string) ([]OAuthClientTok
 		go r.store.RemoveSortedSetRange(key, "-inf", cleanupStartScore)
 	}
 
+	itemsPerPage := config.Global().PaginationItemsPerPage
+
+	if (len(tokens)) == 0 {
+		return []OAuthClientToken{}, 0, nil
+	}
+
+	startIdx, endIdx := 0, itemsPerPage
+	if page > 1 {
+		startIdx = (page - 1) * itemsPerPage
+		endIdx += startIdx
+
+		// Make sure an "out of range" error never happens
+		n := len(tokens)
+		if endIdx > n {
+			endIdx = n
+		}
+
+		if startIdx > n {
+			startIdx = n
+		}
+	}
+
+	totalPages := math.Ceil(float64(len(tokens)) / float64(itemsPerPage))
+
+	tokens = tokens[startIdx:endIdx]
```

Comment:
I am fine with this implementation but in case when we have many tokens  - I do have concerns because here we still pull over network all elements (note how we do `"+inf"`)

Might worth to consider taking advantage of sorted set - being able to move "window" easily using items score. Say we do `r.store.GetSortedSetRange(key, startScore, endScore)` where by default `endScore` is `startScore` plus some range like one day and then we go like this until we fill the page of required size.

We could also count number of elements in sorted set with `ZCOUNT initial_start_score +inf` to get number of possible pages

In that case client would need to receive `next_start_with` and provide it for next page (instead of page number), `next_start_with` is just some value with encoded in it `startScore` + `token` which are last used `endScore` + `token` on previous page. If we do so we will know from where to move "window" in our sorted set for next page.

It might sound not trivial.. just thinking about reducing network I/O and offloading Redis